src/matter/interaction/InteractionMessenger.ts

```diff
@@ -153,8 +153,14 @@ export class InteractionServerMessenger extends InteractionMessenger<MatterDevic
                 }
             }
         } catch (error: any) {
-            logger.error(error.stack ?? error);
-            await this.sendStatus(StatusCode.Failure);
+            let statusCode = StatusCode.Failure;
```

Comment:
Quite hard to read. Maybe if (error instanceof StatusResponseError) and inline both code paths?